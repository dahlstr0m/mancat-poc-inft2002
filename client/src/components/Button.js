// @flow

import * as React from 'react';
import { Component } from 'react-simplified';

/**
 * Renders a success button using Bootstrap styles.
 */
class ButtonSuccess extends Component<{
  onClick: () => mixed,
  small?: boolean,
  children?: React.Node,
}> {
  render() {
    return (
      <button
        type="button"
        className={'btn btn-success' + (this.props.small ? ' btn-sm py-0' : '')}
        onClick={this.props.onClick}
      >
        {this.props.children}
      </button>
    );
  }
}

/**
 * Renders a danger button using Bootstrap styles.
 */
class ButtonDanger extends Component<{
  onClick: () => mixed,
  small?: boolean,
  children?: React.Node,
}> {
  render() {
    return (
      <button
        type="button"
        className={'btn btn-danger' + (this.props.small ? ' btn-sm py-0' : '')}
        onClick={this.props.onClick}
      >
        {this.props.children}
      </button>
    );
  }
}

/**
 * Renders a light button using Bootstrap styles.
 */
class ButtonLight extends Component<{
  onClick: () => mixed,
  small?: boolean,
  children?: React.Node,
}> {
  render() {
    return (
      <button
        type="button"
        className={'btn btn-light' + (this.props.small ? ' btn-sm py-0' : '')}
        onClick={this.props.onClick}
      >
        {this.props.children}
      </button>
    );
  }
}

/**
 * Renders a button using Bootstrap styles.
 */
export default class Button {
  static Success = ButtonSuccess;
  static Danger = ButtonDanger;
  static Light = ButtonLight;
}
