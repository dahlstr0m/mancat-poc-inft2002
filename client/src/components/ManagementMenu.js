// @flow

import * as React from 'react';
import { Component } from 'react-simplified';
import { history } from '../index';
import Button from './Button';
import { CardGrid, CardColumn, CardBody, Card } from './Card';
import { Alert } from './Widgets';

/**
 * Renders All management menu .
 */
export default class ManagementMenu extends Component {
  render() {
    return (
      <CardGrid columns={1} columnsSm={2} columnsMd={3} columnsLg={4}>
        <CardColumn>
          <CardBody title="Projects">
            <Card title="Add new">
              <Button.Success onClick={() => history.push('/admin/projects/new')}>
                Create
              </Button.Success>
            </Card>
          </CardBody>
          <CardBody>
            <Card title="Rearrange order">
              <Button.Light
                onClick={() => Alert.info('This feature is not yet fully implemented. ')}
              >
                Manage
              </Button.Light>
            </Card>
          </CardBody>
        </CardColumn>
        <CardColumn>
          <CardBody title="Poster">
            <Card title="Add new poster">
              <Button.Success onClick={() => history.push('/admin/posters/new')}>
                Create
              </Button.Success>
            </Card>
          </CardBody>
          <CardBody>
            <Card title="Existing posters">
              <Button.Danger onClick={() => history.push('/admin/posters/manage')}>
                Edit
              </Button.Danger>
            </Card>
          </CardBody>
        </CardColumn>
        <CardColumn>
          <CardBody title="Employers">
            <Card title="Manage employers">
              <Button.Danger onClick={() => history.push('/admin/employers')}>Edit</Button.Danger>
            </Card>
          </CardBody>
        </CardColumn>
        <CardColumn>
          <CardBody title="Categories">
            <Card title="Manage Categories">
              <Button.Danger onClick={() => history.push('/admin/categories')}>Edit</Button.Danger>
            </Card>
          </CardBody>
        </CardColumn>
      </CardGrid>
    );
  }
}
