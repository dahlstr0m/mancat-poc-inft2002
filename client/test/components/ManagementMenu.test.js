// @flow

import * as React from 'react';
import { shallow } from 'enzyme';

// Components
import ManagementMenu from '../../src/components/ManagementMenu';
import Button from '../../src/components/Button';

// Tests
describe('ManagementMenu component tests', () => {
  test('ManagementMenu draws correctly', (done) => {
    const wrapper = shallow(<ManagementMenu />);

    // Wait for events to complete
    setTimeout(() => {
      setTimeout(() => {
        expect(wrapper).toMatchSnapshot();

        done();
      });
    });
  });

  test('New project button sets location correctly', (done) => {
    const wrapper = shallow(<ManagementMenu />);

    wrapper.find(Button.Success).at(0).simulate('click');
    setTimeout(() => {
      expect(location.hash).toEqual('#/admin/projects/new');
      done();
    });
  });

  test('New poster button sets location correctly', (done) => {
    const wrapper = shallow(<ManagementMenu />);

    wrapper.find(Button.Success).at(1).simulate('click');
    setTimeout(() => {
      expect(location.hash).toEqual('#/admin/posters/new');
      done();
    });
  });

  test('Edit posters button sets location correctly', (done) => {
    const wrapper = shallow(<ManagementMenu />);

    wrapper.find(Button.Danger).at(0).simulate('click');
    setTimeout(() => {
      expect(location.hash).toEqual('#/admin/posters/manage');
      done();
    });
  });

  test('Manage employers button sets location correctly', (done) => {
    const wrapper = shallow(<ManagementMenu />);

    wrapper.find(Button.Danger).at(1).simulate('click');
    setTimeout(() => {
      expect(location.hash).toEqual('#/admin/employers');
      done();
    });
  });

  test('Manage categories button sets location correctly', (done) => {
    const wrapper = shallow(<ManagementMenu />);

    wrapper.find(Button.Danger).at(2).simulate('click');
    setTimeout(() => {
      expect(location.hash).toEqual('#/admin/categories');
      done();
    });
  });
});
