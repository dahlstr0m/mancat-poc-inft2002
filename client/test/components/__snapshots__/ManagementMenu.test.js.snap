// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ManagementMenu component tests ManagementMenu draws correctly 1`] = `
<CardGrid
  columns={1}
  columnsLg={4}
  columnsMd={3}
  columnsSm={2}
>
  <CardColumn>
    <CardBody
      title="Projects"
    >
      <Card
        title="Add new"
      >
        <ButtonSuccess
          onClick={[Function]}
        >
          Create
        </ButtonSuccess>
      </Card>
    </CardBody>
    <CardBody>
      <Card
        title="Rearrange order"
      >
        <ButtonLight
          onClick={[Function]}
        >
          Manage
        </ButtonLight>
      </Card>
    </CardBody>
  </CardColumn>
  <CardColumn>
    <CardBody
      title="Poster"
    >
      <Card
        title="Add new poster"
      >
        <ButtonSuccess
          onClick={[Function]}
        >
          Create
        </ButtonSuccess>
      </Card>
    </CardBody>
    <CardBody>
      <Card
        title="Existing posters"
      >
        <ButtonDanger
          onClick={[Function]}
        >
          Edit
        </ButtonDanger>
      </Card>
    </CardBody>
  </CardColumn>
  <CardColumn>
    <CardBody
      title="Employers"
    >
      <Card
        title="Manage employers"
      >
        <ButtonDanger
          onClick={[Function]}
        >
          Edit
        </ButtonDanger>
      </Card>
    </CardBody>
  </CardColumn>
  <CardColumn>
    <CardBody
      title="Categories"
    >
      <Card
        title="Manage Categories"
      >
        <ButtonDanger
          onClick={[Function]}
        >
          Edit
        </ButtonDanger>
      </Card>
    </CardBody>
  </CardColumn>
</CardGrid>
`;
